test/unit/submission/validation.spec.ts

```diff
@@ -23,3207 +23,3208 @@ import _ from 'lodash';
+			ClinicalSubmissionRecordsOperations.addRecord(
+				ClinicalEntitySchemaNames.TREATMENT,
+				submissionRecordsMap,
+				{
+					[SampleRegistrationFieldsEnum.submitter_donor_id]: 'ICGC_0002',
+					[ClinicalUniqueIdentifier[ClinicalEntitySchemaNames.FOLLOW_UP]]: 'FLL1234',
+					[ClinicalUniqueIdentifier[ClinicalEntitySchemaNames.TREATMENT]]: 'T_02',
+					[TreatmentFieldsEnum.treatment_type]: ['Chemotherapy'],
+					[TreatmentFieldsEnum.submitter_primary_diagnosis_id]: 'P4',
+					[TreatmentFieldsEnum.treatment_start_interval]: 40,
+					index: 0,
+				},
+			);
+
+			const result = await dv
+				.validateSubmissionData({ ICGC_0002: submissionRecordsMap }, { ICGC_0002: existingDonor })
+				.catch((err: any) => fail(err));
+
+			const error_1: SubmissionValidationError = {
+				fieldName: TreatmentFieldsEnum.treatment_start_interval,
+				message: 'treatment_start_interval cannot be greater than FollowUp interval_of_followup.',
+				type: DataValidationErrors.TREATMENT_TIME_CONFLICT,
+				index: 0,
+				info: {
+					donorSubmitterId: 'ICGC_0002',
+					value: 40,
+				},
+			};
+
+			const error_2: SubmissionValidationError = {
+				fieldName: TreatmentFieldsEnum.treatment_start_interval,
+				message: 'treatment_start_interval should be less than Donor survival_time.',
+				type: DataValidationErrors.TREATMENT_DONOR_TIME_CONFLICT,
+				index: 0,
+				info: {
+					donorSubmitterId: 'ICGC_0002',
+					value: 40,
+				},
+			};
+
+			chai.expect(result[ClinicalEntitySchemaNames.TREATMENT].dataErrors[0]).to.deep.eq(error_2);
+			chai.expect(result[ClinicalEntitySchemaNames.TREATMENT].dataErrors[1]).to.deep.eq(error_1);
+		});
+
+		/**
+		 * Surgery Tests
+		 */
+		it('should allow submitting surgeries with the same combo of submitter_donor_id and sumitter_treatment_id when surgery_type is the same ', async () => {
+			const existingDonor = stubs.validation.existingDonor09();
+			const submissionRecordsMap = {};
+			ClinicalSubmissionRecordsOperations.addRecord(
+				ClinicalEntitySchemaNames.SURGERY,
+				submissionRecordsMap,
+				{
+					[SampleRegistrationFieldsEnum.submitter_donor_id]: 'ICGC_0002',
+					[SurgeryFieldsEnum.submitter_specimen_id]: 'sp-2',
+					[TreatmentFieldsEnum.submitter_treatment_id]: 'Tr-1',
+					[SurgeryFieldsEnum.surgery_type]: 'Biopsy',
+					index: 0,
+				},
+			);
+
+			const result = await dv
+				.validateSubmissionData({ ICGC_0002: submissionRecordsMap }, { ICGC_0002: existingDonor })
+				.catch((err: any) => fail(err));
+
+			chai.expect(result[ClinicalEntitySchemaNames.SURGERY].dataErrors.length).to.equal(0);
+		});
+
+		it('should error when multiple surgeries with the same submitter_specimen_id are found', async () => {
+			const existingDonor = stubs.validation.existingDonor09();
+			const submissionRecordsMap = {};
+			ClinicalSubmissionRecordsOperations.addRecord(
+				ClinicalEntitySchemaNames.SURGERY,
+				submissionRecordsMap,
+				{
+					program_id: 'TEST-CA',
```

Comment:
program_id added to make records duplicate eachother
Majority of changes in this file are just linting

Reply:
PR to Lint: https://github.com/icgc-argo/argo-clinical/pull/1163